core/src/test/scala/doric/syntax/DateColumnsSpec.scala

```diff
@@ -1,17 +1,83 @@
 package doric
 package syntax
 
+import doric.types.{DateType, SparkType}
+
 import java.sql.{Date, Timestamp}
 import java.time.{Instant, LocalDate}
 import org.scalatest.EitherValues
 import org.scalatest.matchers.should.Matchers
+import org.apache.spark.sql.{DataFrame, functions => f}
+import org.scalatest.funspec.AnyFunSpecLike
+
+import scala.reflect.{ClassTag, classTag}
+
+trait DateOperationsSpec extends AnyFunSpecLike with TypedColumnTest {
+
+  def df: DataFrame
+
+  def test[T: DateType: SparkType: ClassTag](tagName: String): Unit = {
+
+    describe(s"Date type $tagName") {
+      it(">") {
+        test[T, T, Boolean](tagName, (a, b) => a > b)
+      }
+      it(">=") {
+        test[T, T, Boolean](tagName, (a, b) => a >= b)
+      }
+      it("<") {
+        test[T, T, Boolean](tagName, (a, b) => a < b)
+      }
+      it("<=") {
+        test[T, T, Boolean](tagName, (a, b) => a <= b)
+      }
+    }
+  }
 
-import org.apache.spark.sql.{functions => f}
+  def test[T1: SparkType: ClassTag, T2: SparkType: ClassTag, O: SparkType](
+      tagName: String,
+      f: (DoricColumn[T1], DoricColumn[T2]) => DoricColumn[O]
+  ): Unit =
+    df.validateColumnType(
+      f(
+        col[T1](getName(tagName)),
+        col[T2](getName(tagName))
+      )
+    )
+
+  def getName(tagName: String, pos: Int = 1): String =
+    s"col_${tagName}_$pos"
```

Comment:
I originally planned to use `classTag` here, but that always gave back `GenericClassTag` so I chose using string names instead. If anyone has any ideas on alternatives I would love to know them!

Reply:
I did, but it kept returning `GenericClassTag` for each of the columns which when Spark couldn't understand as it couldn't tell the columns apart. I'm not certain on this but perhaps because `Instant`, `Date`, `LocalDate` and `Timestamp` all come from Java it doesn't quite know how to get a class tag for them? 🤔

Reply:
Leave it as it is, the only way I'm seeing to do it automatically is by collecting one of the values of the dataframe and I don't believe is worth it.